tests/RequestMatcher/ClosureRequestMatcher.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Kevinrob\GuzzleCache\Tests\RequestMatcher;
+
+use Closure;
+use Kevinrob\GuzzleCache\Strategy\Delegate\RequestMatcherInterface;
+use Psr\Http\Message\RequestInterface;
+
+class ClosureRequestMatcher implements RequestMatcherInterface
+{
+    /**
+     * @var Closure
+     */
+    private $closure;
+
+    /**
+     * CallbackRequestMatcher constructor.
+     */
+    public function __construct(Closure $closure)
```

Comment:
Ho right. I don't see that it's only for tests!
I am influenced by functional programming... So my thoughts are biased. I trust you.